### PR TITLE
Basic support for Confluent's  KafkaAvroSerializer / KafkaAvroDeSerializer

### PR DESCRIPTION
**Pull request by fmck3516**
Basic support to serialize and deserialize with Confluent's [proprietary 5-byte prefix](https://docs.confluent.io/platform/current/schema-registry/serdes-develop/index.html#wire-format).

The prefix will be added if the key or value serializer is set to `io.confluent.kafka.serializers.KafkaAvroSerializer`. 
The prefix will be removed if the key or value deserializer is set to `io.confluent.kafka.serializers.KafkaAvroDeserializer`.

**Configuration example:**
```
var configuration = JSON.stringify({
    consumer: {
        keyDeserializer: "io.confluent.kafka.serializers.KafkaAvroDeserializer",
        valueDeserializer: "io.confluent.kafka.serializers.KafkaAvroDeserializer",
    },
    producer: {
        keySerializer: "io.confluent.kafka.serializers.KafkaAvroSerializer",
        valueSerializer: "io.confluent.kafka.serializers.KafkaAvroSerializer",
    },
    schemaRegistry: {
        url: "https://subdomain.us-east-2.aws.confluent.cloud",
        basicAuth: {
            credentialsSource: "USER_INFO",
            userInfo: "KEY:SECRET"
        },
    },
})
```

The configuration for `schemaRegistry` is required to use a serializer of type `io.confluent.kafka.serializers.KafkaAvroSerializer`.

A new `produce` and `consume` function has been added (each accepts the new configuration argument):

`produceWithConfiguration(producer, messages, configuration, keySchema, valueSchema);`
`consumeWithConfiguration(consumer, 20, configuration, keySchema, valueSchema);`

**Note**: I ported a sub-set of the existing functionality of the Confluent Java client. This code will be obsolete once Confluent adds Schema Registry support for their Go client.  At the time of writing, the Go client does not support Schema Registry: https://docs.confluent.io/platform/current/clients/index.html

### CODE REVIEW

**Comment by fmck3516** [comment]
I'm not familiar with Go - I still have to figure out how to unit test the stuff.

File: test/test.js
Comment (mostafa) [inline test/test.js:1]: I am wondering if you could move all the scripts (`test_*.js`) to the `scripts` directory and rename this somehow to something like `test_avro_with_magic_bytes.js` or something shorter.

**Review by mostafa — CHANGES_REQUESTED** [review]
Hey @fmck3516,

Thanks for the awesome PR. :pray: I left a few comments and I think it can be merged after the changes.

**Comment by fmck3516** [comment]
@mostafa: Thanks for the feedback - I will update the PR in the upcoming week.

File: consumer.go
Comment (fmck3516) [reply consumer.go]: Typo - should be `And`. 

> Could it be just `removeMagicByte`?

No, that would be technically incorrect. The magic byte is the leading zero. I need to remove the following 4-bytes for the schema id as well.

File: schemaRegistry.go
Comment (fmck3516) [reply schemaRegistry.go]: I'd rather fail fast in this case. The schema lookup is only performed if the serializer is set to `io.confluent.kafka.serializers.KafkaAvroSerializer`. A message without prefix would violate Confluent's wire format contract.

**Comment by fmck3516** [comment]
I incorporated your feedback. I also switched to a JSON-based configuration model to be in sync with the approach for the `auth` configuration.

What do you think about having one big configuration JSON for auth and everything else?

File: producer.go
Comment (fmck3516) [reply producer.go]: Typo - should be `And`. 

> Could it be just `addMagicByte`?

No, that would be technically incorrect. The magic byte is only the leading zero. I need to add the 4-bytes for the schema id as well.

File: test/test.js
Comment (fmck3516) [reply test/test.js]: I renamed the file to `test_avro_with_schema_registry.js` and yes, I think it would be good to move the tests out of the top level directory.

**Comment by mostafa** [comment]
@fmck3516 I was actually thinking about the same thing, but then thought that it's out of scope of this PR or mine. So maybe the next PR would focus on refactoring, cleanups and better configuration management. I also have the idea to refactor metric collection and error handling. Sorry for too much information! :rofl:

File: test_avro_with_schema_registry.js
Comment (mostafa) [inline test_avro_with_schema_registry.js:57]: I love this. :clap:

File: test/test.js
Comment (mostafa) [reply test/test.js]: But the file is still called `test/test.js`. Am I missing something? :thinking:

**Review by mostafa — APPROVED** [review]
LGTM! I can rearrange test scripts (files) after merging.

**Comment by fmck3516** [comment]
> @fmck3516 I was actually thinking about the same thing, but then thought that it's out of scope of this PR or mine.
That makes sense - one step at a time. 

>  So maybe the next PR would focus on refactoring, cleanups and better configuration management. I also have the idea to refactor metric collection and error handling. Sorry for too much information! 🤣
One thing I would like to add to the wishlist: A suite of regression tests that is executed automatically.

**Pull request merged**
